### PR TITLE
Implement API for user group

### PR DESCRIPTION
**Pull request by taohe1012**
Implement API for user group
 - add CRUD API for user group
 - add add/remove groupMemebr API for user group

# PR Submission checklist

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/888 |

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [ ] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

All unit tests for user group are passed:
```

time="2023-07-13T11:10:48+08:00" level=debug msg="opts.Insecure : 'true'"
time="2023-07-13T11:10:48+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/latest/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:49+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:48 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
=== RUN   TestGroupGet
time="2023-07-13T11:10:49+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/?cached=false&resolve_names=false&query_member_of=false&limit=1000 HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:50+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:49 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:50+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:52+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:50 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
--- PASS: TestGroupGet (2.47s)
=== RUN   TestGroupCreate
time="2023-07-13T11:10:52+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/users/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"name\":\"test_user_group_member\"}\n"
time="2023-07-13T11:10:53+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:52 GMT\n    Location: /1/auth/users/SID%!A(MISSING)S-1-5-21-2193650305-1279797252-961391754-1156\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:53+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/users/USER:test_user_group_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:54+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:53 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:54+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/groups/?force=true HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"gid\":100000,\"members\":[{\"id\":\"UID:2120\",\"type\":\"user\"}],\"name\":\"test_group_create_options\"}\n"
time="2023-07-13T11:10:55+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:54 GMT\n    Location: /1/auth/groups/SID%!A(MISSING)S-1-5-21-2193650305-1279797252-961391754-1157\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:55+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GID:100000 HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:56+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:55 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:56+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/groups/GID:100000 HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:58+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:10:57 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:58+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GID:100000 HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:58+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 404 Not Found\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:57 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:10:58+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/users/USER:test_user_group_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:10:59+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:10:58 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
--- PASS: TestGroupCreate (7.88s)
=== RUN   TestGroupUpdate
time="2023-07-13T11:10:59+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/groups/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"name\":\"test_group_create_update\"}\n"
time="2023-07-13T11:11:00+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:10:59 GMT\n    Location: /1/auth/groups/SID%!A(MISSING)S-1-5-21-2193650305-1279797252-961391754-1158\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:00+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GROUP:test_group_create_update HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:01+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:00 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:01+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    PUT /platform/1/auth/groups/GROUP:test_group_create_update?force=true HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"gid\":10000}\n"
time="2023-07-13T11:11:02+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Content-Length: 0\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:11:01 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "       
time="2023-07-13T11:11:02+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GID:10000 HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:03+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:02 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:03+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/groups/GROUP:test_group_create_update HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:04+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:11:03 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
--- PASS: TestGroupUpdate (4.40s)
=== RUN   TestGroupMemberAdd
time="2023-07-13T11:11:04+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/groups/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"name\":\"test_group_add_member\"}\n"
time="2023-07-13T11:11:04+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:03 GMT\n    Location: /1/auth/groups/SID%!A(MISSING)S-1-5-21-2193650305-1279797252-961391754-1159\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:04+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GROUP:test_group_add_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:05+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:04 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:05+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GROUP:test_group_add_member/members/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:07+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, POST, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:06 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:07+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/users/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"name\":\"test_user_group_add_member\"}\n"
time="2023-07-13T11:11:08+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:07 GMT\n    Location: /1/auth/users/SID%!A(MISSING)S-1-5-21-2193650305-1279797252-961391754-1160\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:08+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    POST /platform/1/auth/groups/GROUP:test_group_add_member/members/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    Content-Type: application/json\n    \n    {\"name\":\"test_user_group_add_member\"}\n"
time="2023-07-13T11:11:09+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 201 Created\n    Transfer-Encoding: chunked\n    Allow: GET, POST, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:08 GMT\n    Location: /1/auth/groups/GROUP:test_group_add_member/members/UID%!A(MISSING)2121\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:09+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GROUP:test_group_add_member/members/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:10+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, POST, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:09 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:10+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/groups/GROUP:test_group_add_member/members/USER:test_user_group_add_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:12+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:11:10 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:12+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    GET /platform/1/auth/groups/GROUP:test_group_add_member/members/ HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:12+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 200 Ok\n    Transfer-Encoding: chunked\n    Allow: GET, POST, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Content-Type: application/json\n    Date: Thu, 13 Jul 2023 03:11:11 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:12+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/users/USER:test_user_group_add_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:13+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:11:12 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
time="2023-07-13T11:11:13+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP REQUEST -------------------------\n    DELETE /platform/1/auth/groups/GROUP:test_group_add_member HTTP/1.1\n    Host: 10.230.24.249:8080\n    Authorization: root:******\n    \n"
time="2023-07-13T11:11:14+08:00" level=debug msg="\n    -------------------------- GOISILON HTTP RESPONSE -------------------------\n    HTTP/1.1 204 No Content\n    Allow: GET, PUT, DELETE, HEAD\n    Content-Security-Policy: default-src 'self' 'unsafe-inline' 'unsafe-eval' data:; script-src 'self' 'unsafe-eval'; style-src 'unsafe-inline' 'self';\n    Date: Thu, 13 Jul 2023 03:11:13 GMT\n    Server: Apache\n    Strict-Transport-Security: max-age=31536000;\n    X-Content-Type-Options: nosniff\n    X-Frame-Options: sameorigin\n    X-Xss-Protection: 1; mode=block\n    "
--- PASS: TestGroupMemberAdd (9.81s)
PASS
ok      github.com/dell/goisilon        28.208s
```

### CODE REVIEW

File: api/v1/api_v1_user_groups.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+		group = groupListResp.Groups[0]
+		return
+	}
+
+	return nil, fmt.Errorf("group not found: %s", authGroupId)
+}
+
+// GetIsiGroupList queries all groups on the cluster with filter,
+// filter by namePrefix, domain, zone, provider, cached, resolveNames, memberOf, zone and limit.
+func GetIsiGroupList(ctx context.Context, client api.Client,
```
Comment (chimanjain) [inline api/v1/api_v1_user_groups.go:51]: can we convert the parameters of this function into a struct

File: user_group.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+// GetAllGroups returns all groups on the cluster.
+func (c *Client) GetAllGroups(ctx context.Context) (GroupList, error) {
+	return c.GetGroupsWithFilter(ctx, nil, nil, nil, nil, nil, nil, nil, nil)
+}
+
+// GetGroupsWithFilter returns groups on the cluster with
+// Optional filter: namePrefix, domain, zone, provider, cached, resolveNames, memberOf, zone and limit.
+func (c *Client) GetGroupsWithFilter(ctx context.Context,
```
Comment (chimanjain) [inline user_group.go:46]: can we convert the parameters of this function into a struct

File: api/v1/api_v1_user_groups.go (hunk 1 above)
Comment (taohe1012) [reply api/v1/api_v1_user_groups.go:51]: @chimanjain, in this way, user just need fill nil in the function; else if covert to struct, user need import this struct and fill it.
And also to keep consistent, other functions are using this way, not using struct.

Comment (chimanjain) [reply api/v1/api_v1_user_groups.go:51]: ok

Comment (taohe1012) [reply api/v1/api_v1_user_groups.go:51]: thanks @chimanjain

**Review by chimanjain — APPROVED** [review]

**Review by shefali-malhotra — APPROVED** [review]
LGTM

**Pull request merged**